biodatasets/medquad/medquad.py

```diff
@@ -0,0 +1,264 @@
+            doc_root = ET.parse(file_path).getroot()
+            document_id = doc_root.attrib.get("id")
+            for element in doc_root:
+                if element.tag == "QAPairs":
+                    qa_pairs = element
+                    for qa_pair in qa_pairs:
+                        # Handled this way in case question & answer order occur differently
+                        question = qa_pair[0] if qa_pair[0].tag == "Question" else qa_pair[1]
+                        answer = qa_pair[1] if qa_pair[1].tag == "Answer" else qa_pair[0]
+
+                        qa_list.append({
+                            "Document": document_id,
+                            "QAPair": qa_pair.attrib.get("pid"),
+                            "qid": question.attrib.get("qid"),
+                            "qtype": question.attrib.get("qtype"),
+                            "Question": question.text,
+                            "Answer": answer.text,
+                        })
+
+        return qa_list, os.path.dirname(file_paths[0])
+
+    def _dump_xml_to_json(self, dl_manager) -> str:
+        """
+        This method parses the dataset
+
+        """
+        if self.config.subset_id == "medquad":
+            file_base_urls = _URLS[f"{self.config.subset_id}_base_urls"]
+            qa_pairs_enriched_fname = f"MedQuADGoldenEnriched/{self.config.subset_id}.json"
+            repo_files = json.loads(requests.get(_DATA_REPO_FETCH_URL).text)
+        else:
+            raise NotImplementedError("Only full set loader is implemented here")
+
+        # Collect path info for all repo paths, and determine relevant XML files
+        qa_file_paths = []
+        for subset_name, uri_ in file_base_urls.items():
+            for path in repo_files["tree"]:
+                if path["type"] == "blob" and path["path"].startswith(uri_) and path["path"].endswith(".xml"):
+                    qa_file_paths.append(os.path.join(_DATA_PATH, path["path"]))
+
+        qa_list, data_dir = self._load_qa_from_xml(
+            file_paths=dl_manager.download_and_extract(qa_file_paths)
+        )
+
+        qa_pairs_enriched_full_path = os.path.join(data_dir, qa_pairs_enriched_fname)
+
+        qa_pairs_enriched_dir = os.path.dirname(qa_pairs_enriched_full_path)
+        if not os.path.exists(qa_pairs_enriched_dir):
+            os.mkdir(qa_pairs_enriched_dir)
+
+        if not os.path.exists(qa_pairs_enriched_full_path):
+            data = {"qa_pairs": qa_list}
+            # dump QA paris to json
+            with open(qa_pairs_enriched_full_path, "wt", encoding="utf-8") as file:
+                json.dump(data, file, indent=2)
+
+        return qa_pairs_enriched_full_path
+
+    def _split_generators(self, dl_manager) -> List[datasets.SplitGenerator]:
+        """Returns SplitGenerators."""
+
+        qa_pairs_enriched_fpath = self._dump_xml_to_json(dl_manager)
+
+        # There is no canonical train/valid/test set in this dataset. So, only TRAIN is added.
```

Comment:
I think they use this for testing: https://github.com/abachaa/LiveQA_MedicalTask_TREC2017/tree/master/TestDataset

Reply:
I added this set, but the general scheme doesn't match, I implemented it to barely match the schema.